server/user.go

```diff
@@ -295,18 +313,30 @@ func (s *UserMgmtServer) getCreds(r *http.Request, requiresAdmin bool) (api.Cred
 		return api.Creds{}, err
 	}
 
-	isAdmin, err := s.cm.IsDexAdmin(clientID)
-	if err != nil {
-		log.Errorf("userMgmtServer: GetCreds err: %q", err)
-		return api.Creds{}, err
+	i := 0
+	for _, clientID := range clientIDs {
+		// Make sure the client actually exists.
+		isAdmin, err := s.cm.IsDexAdmin(clientID)
+		if err != nil {
+			log.Errorf("userMgmtServer: GetCreds err: %q", err)
+			return api.Creds{}, err
+		}
+		// If the endpoint requires an admin client, filter out clients which are not admins.
+		if isAdmin || !requiresAdmin {
+			clientIDs[i] = clientID
+			i++
+		}
 	}
-	if requiresAdmin && !isAdmin {
+
+	clientIDs = clientIDs[:i]
```

Comment:
The check on line 278 is verifying the token is signed by dex, has the correct issuer, isn't expired, etc. This is a bit different, it's checking if the client itself has "admin" status. I think agree that the logic could be cleaned up a bit.

So maybe, a better comment on line 278, and more obvious logic in this chunk?